CRM/Utils/Mail.php

```diff
@@ -199,6 +199,9 @@ public static function send(array &$params): bool {
       // This is copied from the Action Schedule send code.
       $textMessage = str_replace('&amp;', '&', $textMessage);
     }
+    if (str_contains($textMessage, 'Undefined array key') || str_contains($htmlMessage, 'Undefined array key') || str_contains($htmlMessage, 'Undefined index')) {
+      CRM_Core_Error::deprecatedWarning('email output affected by undefined php properties:' . CRM_Utils_String::purifyHTML($htmlMessage));
```

Comment:
we could only give the html for unit tests I guess - to prevent the logs getting it - but then it might be hard to pin down

Reply:
It's just that even with just the warning a civimail to 10000 people with some smarty in it where a notice comes up would log it 10000 times.

Does it reliably have a workflow_name it can look at here, and only log it for system message templates unless opt-in for all? Or some idea like that?

Reply:
I don't think there is a workflow_name at that point - I suppose we could use civi statics to only log once per php process (once per unit test in the case of tests).  I feel like a notice in a civimail would be kinda bad too so wouldn't want to entirely ignore it